### ISSUE TITLE
[PATCH] Allow more choice with early techs and provide a tutorial

### ISSUE THREAD

**Issue by ischwarze** [freeorion/freeorion#2317]
The idea of the following patch is to:

1. Allow more choice for the player during the earliest phase of the game,
more efficiently using the existing technologies, buildings, and ship
components for actually playing with them in FreeOrion style, rather than
just getting them revealed up front, with no choice to be made.

2. Provide an integrated tutorial.  This is intricately linked to goal 1
above.  When game concepts are revealed one by one, an integrated tutorial
arises very naturally.  When everything is available from the start,
the player needs to use all of the interface in turn 1, and integrating a
tutorial becomes very hard.

I am well aware that sending a patch of this size implies a considerable
risk - and you are saying just that on your website.  But checking whether 
the concept works really required a proof-of-concept implementation,
having an abstract design discussion wouldn't have provided the same
amount of insight, and design and implementation really cross-fertilized
each other.  Besides, you can now play test games to judge the idea
rather than just considering it theoretically and merely *guessing* how it might
feel.

If you like the general direction (or parts of it), i see two options 
to get on with it:

A. Either you tell me which aspects still require polishing, i fix those,
such that you can put it in as one comprehensive patch.  Least work for
you if you consider that viable.

B. Or you tell me which aspects you like, and specify an order, and i send 
individual, smaller, incremental patches to be committed one by one.

Even though the size of the patch is not small, it closely sticks to the
existing FreeOrion concepts, technologies, and objects, merely better 
exploiting them in the starting phase - and a substatial part of it is
tutorial text.  I already enjoyed playing a couple of test games with 
these enhancements, and my experience is that strategies become more
varied and in particular more markedly different for research-focussed
versus industry-focussed species and for starting surroundings with
different properties.  All the same, with the minimal adjustments 
included in the patch, the AI deals with it pretty well, following 
the classic strategy, which is still completely viable though maybe
not totally optimal in some situations - but the AI is hardly playing 
an optimal strategy anyway, but rather a very solid and robust one.

To give just one example of how differently games can evolve, depending on the situation: in one test game, i played a "colonization rush" strategy and had the complete tutorial (all seven parts) finished before turn 20.  In another test game, i played a military strategy, capturing three nearby native planets and one nearby planet belonging to an AI player first and only reached tutorial step 06 in turn 60 and tutorial step 07 in turn 70 - yet by turn 75, i had four inhabited colonies and three additional outposts in the process of constructing colony buildings.  In a third test game, i was too careless with defense and got myself rushed and killed by an AI player who arrived very early with exactly the right amount of warships and troops.  So this really helps to improve the variety of strategies and game developments in the beginning of the game.

Thoughts?
Ingo

P.S.
If you want to head into this direction, this patch obsoletes the 
bugtracker entry #2304 (improve SITREP_WELCOME, GREETINGS_GUIDE_TEXT,
QUICK_START_GUIDE_TEXT).

[0001-Allow-more-choice-with-early-techs-and-provide-a-tut.patch.txt](https://github.com/freeorion/freeorion/files/2559503/0001-Allow-more-choice-with-early-techs-and-provide-a-tut.patch.txt)

**Comment by Dilvish-fo** [comment]
Hmm.  I thought to myself, well, rather than press ischwarz to explain more about this perplexing principle that leads him to post patches in Issues but won't let him make a PR, I'll just make a branch for his patch, apply it and commit, and make a PR for him, so that others could more easily consider and test his ideas.  

But upon doing a ```git apply``` with this patch and current master (@ischwarze if there is some other way I am supposed to apply this please explain), I note a couple oddities that significantly impede that plan-- (i) there seem to be a great many new files added by the patch but they don't wind up actually being added as being tracked by git, so to to add them to git I would need to list all the various untracked files in the source tree and filter out and add tracking for the ones that were created  by this patch without accidentally adding any of the multitude of properly untracked (various build artifacts I think) files in the source tree; and (ii) ok, well I think I figured this one out, the renaming and later editing of "REMOTE_COL.disabled' seems to have left a weird artifact from git's point of view, but I think I could straighten that out and then it just becomes another instance of probem (i).

So, while that problem is by no means insurmountable, it's more work than I really had in mind to do and have time for just now (it seems it would take quite a bit of care, what with me having no advance idea of just what to look for).

@ischwarze, since you seemed to indicate previously (and the patch seems to indicate) that you have no issue working with git, and it is just some github features that you object to, I would expect that someplace you likely have a git branch with all this code.  Even if you don't want to submit a PR for this, could you perhaps share a link to that branch so that I could just do a ```git pull``` directly from it?  And if it's not currently at a publicly available location, and you don't want to put it on some ideologically-compatible git repo hosting site, then perhaps you could just upload a git bundle for us instead of these patches?

**Comment by Dilvish-fo** [comment]
> But since you apparently have some trouble using raw git that i can't understand (i applied my patch with "git am"...

As I noted, I had tried applying it with 'git apply' and invited you to let me know a better way to apply it, as you've now done (I had seen a reference to 'git am' and wondered if that might be it, but now at least I think I'll know how to recognize a patch formatted for 'git am').

But anyways, thanks much for humoring us and submitting the PR.  Years ago when I first joined the project we just dealt with patches submitted to forum threads, much like your patch attached here, but we have found the github PR format to really assist with reviewing and refining submissions.
